packages/lib/src/vis/tiles/utils.ts

```diff
@@ -0,0 +1,44 @@
+import { Vector2 } from 'three';
+
+import type { Size } from '../models';
+
+export function getTilesOffset(
+  offset: Vector2,
+  size: Size,
+  tileSize: Size
+): Vector2[] {
+  const xNbTiles = Math.ceil(
+    ((offset.x % tileSize.width) + size.width) / tileSize.width
+  );
+  const yNbTiles = Math.ceil(
+    ((offset.y % tileSize.height) + size.height) / tileSize.height
+  );
+
+  const xStart = Math.floor(offset.x / tileSize.width) * tileSize.width;
+  const yStart = Math.floor(offset.y / tileSize.height) * tileSize.height;
+
+  const tilesOffset: Vector2[] = [];
+  for (let yindex = 0; yindex < yNbTiles; yindex += 1) {
+    for (let xindex = 0; xindex < xNbTiles; xindex += 1) {
+      tilesOffset.push(
+        new Vector2(
+          xStart + xindex * tileSize.width,
+          yStart + yindex * tileSize.height
+        )
+      );
+    }
+  }
+
+  return tilesOffset;
+}
+
+export function tileToCenterSquareDistance(
+  offset: Vector2,
+  center: Vector2,
+  tileSize: Size
+): number {
+  return (
+    (offset.x + tileSize.width / 2 - center.x) ** 2 +
```

Comment:
Just an idea: would it make things easier to work with `offset` being the center of the tile rather than the corner ? 

This would avoid the `+tileSize.width / 2` at several places.

Reply:
Probably, I'll try it.
I'm used to place images with the lower-left corner, that's why I did that, but since three uses the center, it should indeed be better.

Reply:
There are some issues with it:
- The API `get` function works with an offset and to me it is best this way because it maps easily to array slicing... so back to `-tileSize.width/2`
- The tiles on the edges are smaller than `tileSize`, so, while the `offset` remains the same, the `center` changes. This specific case would need to be handled when generating the tile centers. The current way to handle this is to use the size of the received array rather than the tile size.

In the end, it removes some offsets at some places, but adds them at other places.

Reply:
Nevermind then. Do as you see fit